chcemvediet/apps/accounts/models.py

```diff
@@ -88,5 +104,32 @@ def has_undecided_emails(self):
         else:
             return self.undecided_emails_set.exists()
 
+    def clean(self):
+        if not self.custom_anonymized_strings:
+            self.custom_anonymized_strings = None
+            return
+        if type(self.custom_anonymized_strings) != list:
+            raise ValidationError(u'JSON must be an array of strings')
+        for line in self.custom_anonymized_strings:
+            if type(line) != unicode:
+                raise ValidationError(u'JSON must be an array of strings')
+
+    def save(self, *args, **kwargs):
+        self._delete_outdated_attachments()
+        super(Profile, self).save(*args, **kwargs)
+        self.__original_custom_anonymized_strings = self.custom_anonymized_strings
+
+    def _delete_outdated_attachments(self):
+        if self._diff_custom_anonymized_strings():
+            AttachmentFinalization.objects.owned_by(self.user).delete()
+            AttachmentAnonymization.objects.owned_by(self.user).delete()
+
+    def _diff_custom_anonymized_strings(self):
```

Comment:
@viliambalaz premenuj to na `_custom_anonymized_strings_changed`. Lebo realne nevracias "diff", ale len boolean ci sa zmenili alebo nie